lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -209,6 +209,17 @@ sub complex_query {
         push @{$attrs{prefetch}}, 'children';
     }
 
+    if ($args{failed_modules}) {
+        push @joins, "modules";
+        push(
+            @conds,
+            {
+                'modules.name' => {-in => $args{failed_modules}},
+                'modules.result' =>
+                  {-in => [OpenQA::Schema::Result::Jobs::FAILED, OpenQA::Schema::Result::Jobs::SOFTFAILED]},
```

Comment:
For soft failed jobs we don't show the modules on the tests overview page so how would it look like when trying to filter for softfailed module "foo". Will it just show orange bubbles without any modules next to it? When in doubt, just don't do anything about softfailed, only handle failed.